### PR TITLE
SALTO-3143: Add shared deploy config to handle missing users

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add deploy config with user fallback strategy

---

This PR contains only the adapter-components changes, for full context see https://github.com/salto-io/salto/pull/3846/files

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

**Review by alonstern — COMMENTED** [review]
LGTM 🔥 
@netama Do you want to have a look too?

**Comment by alonstern** [comment]
Actually one small comment, could you add `"##DEPLOYER##"` in adapter components? Since we are supposed to use it in both adapters

File: packages/adapter-components/src/config/shared.ts [hunk 1]
```diff
@@ -55,6 +57,10 @@ export type UserFetchConfig<T extends Record<string, unknown> | undefined = unde
   hideTypes?: boolean
 }
 
+export type UserDeployConfig = {
```
Comment (netama) [inline packages/adapter-components/src/config/shared.ts:60]: please add a comment explaining what this should be used for (I know we didn't do it consistently enough here in the past...)

**Review by netama — APPROVED** [review]
LGTM 🎈 please add a test for the new function

**Pull request merged**